### ISSUE TITLE
[Enhancement][Windows] Implement direct D3D11VA/DXVA2 video rendering

### ISSUE THREAD

**Issue by alexmercerind** [alexmercerind/media_kit#296]
### Introduction

At present state, [ANGLE (Almost Native Graphics Layer Engine)](https://github.com/google/angle) is used to render videos on Windows. The underlying [libmpv core offers a GL-based rendering API](https://github.com/mpv-player/mpv/blob/master/libmpv/render_gl.h), which is used through OpenGL ES 2.0 implementation provided by ANGLE (& then interop-ed to consume it as D3D texture for access in Flutter).

This method is hardware-accelerated (GPU powered) & decently performant, making package:media_kit able to play multiple 1080p/4K/8K videos at once. Obviously, this depends upon the hardware specification as well (& we can't defy science). Current video-playback performance is very comparable to that in Google-Chrome/Chromium.

### Scope

However, there is still scope for improvement & CPU consumption can be dropped to a literal 0.0%. A number of video output drivers exist in mpv, [including D3D11VA](https://github.com/mpv-player/mpv/blob/master/video/out/d3d11/hwdec_d3d11va.c). It is very efficient but is not exposed as part of libmpv. It should be possible to use this in package:media_kit & further enhance the performance. 

Since this is not present in libmpv by default, one will have to implement it there & then use it here. I'll be very glad to experiment & do it myself. One approach can be to have a new render API much like GL one. Another (simpler) approach can be to expose necessary references (e.g. `ID3D11Texture2D`/`HANDLE` etc. etc.) as libmpv properties & let libmpv run the render loop & do the rendering itself [much like how it's implemented for Android]. I haven't looked at the implementation challenges at the moment since there are other higher-priority issues to work with while the current video rendering performance on Windows is already good.

<hr>

**NOTE:** This has no relation with the fallback/optional software-rendering _i.e._ enableHardwareAcceleration: false.

**Comment by sid-max1996** [comment]
@alexmercerind I play a 4K VP9 video in Windows 10 and have a CPU load of 20% and GPU load of 90%. When I set the configuration for VideoController (hwdec: 'd3d11va', vo: 'gpu'), the CPU load drops to 7% and GPU load to 25%, but the video plays in a separate window.
```dart
  late final controller = VideoController(player,
      configuration: VideoControllerConfiguration(hwdec: 'd3d11va', vo: 'gpu'));
```
![Image](https://github.com/user-attachments/assets/ed1c01c5-1304-413e-b969-158cbb5ccd27)

**Comment by alexmercerind** [comment]
@sid-max1996, yes that is what we want to solve, it should render into Direct3D context from Flutter.

**Comment by sid-max1996** [comment]
> [@sid-max1996](https://github.com/sid-max1996), yes that is what we want to solve, it should render into Direct3D context from Flutter.

That's great — thanks for your work! By the way, if it’s not a secret, how far along is the optimization at now?

**Comment by alexmercerind** [comment]
> By the way, if it’s not a secret, how far along is the optimization at now?

I am not actively working on this project anymore.
